packages/react-hooks/useRelayClient.ts

```diff
@@ -0,0 +1,47 @@
+import { useWalletClient } from "wagmi";
+import { privateKeyToAccount } from "viem/accounts";
+import { getLogger } from "@logtape/logtape";
+import { skipToken, useQuery } from "@tanstack/react-query";
+
+import { RelayClient } from "@massmarket/client";
+
+import { useKeycard } from "./useKeycard.ts";
+import { useRelayEndpoint } from "./useRelayEndpoint.ts";
+import { useShopId } from "./useShopId.ts";
+
+const logger = getLogger(["mass-market", "frontend", "useRelayClient"]);
+
+/**
+ * This hook instantiates a RelayClient class and connects and authenticates the relay.
+ * This query will run on every app session.
+ * Since useQuery caches data in memory during a single app session, it will return the RelayClient class.
+ * However, during refreshes (or when the app unmounts), data is cached in the configured persister (localStorage), and since the class cannot be serialized the query is not cached.
```

Comment:
this is somewhat incorrect. JSON.stringify will return a stringified json obj with the public proprieties of the relayClient, which would then get stored. In the RelayClient class we need to define a toJSON method that returns undefined.  The JSON.stringify(cleint) will return undefined which will cause it to be removed from the prested store https://tanstack.com/query/latest/docs/framework/react/plugins/persistQueryClient#removal